JumpscaleLib/sal/zerotier_controller/ZeroTierController.py

```diff
@@ -0,0 +1,175 @@
+from Jumpscale import j
+import json
+import requests
+import netaddr
+
+BASE_API = "http://127.0.0.1:9993"
+
+# ZT EtherType constants as defined here: https://github.com/zerotier/ZeroTierOne/blob/master/node/Constants.hpp
+ZT_ETHERTYPE_IPV4 = "0x0800"
+ZT_ETHERTYPE_ARP = "0x0806"
+ZT_ETHERTYPE_IPV6 = "0x86dd"
+RULES = [
+    {
+        "etherType": int(ZT_ETHERTYPE_IPV4, 16),
+        "not": True,
+        "or": False,
+        "type": "MATCH_ETHERTYPE"
+    },
+    {
+        "etherType": int(ZT_ETHERTYPE_IPV6, 16),
+        "not": True,
+        "or": False,
+        "type": "MATCH_ETHERTYPE"
+    },
+    {
+        "etherType": int(ZT_ETHERTYPE_ARP, 16),
+        "not": True,
+        "or": False,
+        "type": "MATCH_ETHERTYPE"
+    },
+    {
+        "not": False,
+        "or": False,
+        "type": "ACTION_DROP"
+    },
+    {
+        "not": False,
+        "or": False,
+        "type": "ACTION_ACCEPT"
+    }
+]
+
+
+class ZeroTierController():
+    def __init__(self):
+        self.__jslocation__ = "j.sal.zerotier_Controller"
+        self.controller_status()
+
+    def get_filepath(self):
+        """Get filepath """
+        return "/var/lib/zerotier-one"
+
+    def _request(self, url, payload=None, method="get"):
+        """Simple request wrapper
+        Takes a couple of variables and wraps around the requests
+        module
+        Args:
+            url: API URL
+            method: Query method (default: {"get"})
+            payload: JSON payload (default: {None})
+        Returns:
+            Dataset as result from query
+            JSON Object
+        """
+        header = {"X-ZT1-Auth": self.get_authtoken}
+        req = None
+        if payload is not None:
+            req = requests.post(
+                BASE_API+url, headers=header, json=payload)
+        elif method == "get":
+            req = requests.get(
+                BASE_API+url, headers=header)
+        elif method == "delete":
+            req = requests.delete(
+                BASE_API+url, headers=header)
+        return req.json()
+
+    def controller_status(self):
+        return self._request("/controller", {})
+
+    @property
+    def get_authtoken(self):
```

Comment:
let's be more consistent with naming.... most of the SAL is written with verb last. I suggest `auth_token_get`
```suggestion
    def auth_token_get(self):
```